FWCore/src/components/PileupHitMergeTool.cpp

```diff
@@ -51,26 +51,29 @@ StatusCode PileupHitMergeTool<Hits, PositionedHits>::readPileupCollection(podio:
   }
 
   /// as above, for the positioned collection
-  bool posHitCollectionPresent = store.get(m_pileupPosHitsBranchName, posHitCollection);
-  if (posHitCollectionPresent) {
-    m_posHitCollections.push_back(posHitCollection);
-  } else {
+  if (!store.get(m_pileupPosHitsBranchName, posHitCollection)) {
```

Comment:
I don't really know what's the best way to do it, but maybe it's good to be more explicit about the user intention? If I made an error with the collectionName I maybe would want the program to even exit with failure at this point. But if I don't want to add a signal collection then I don't need to be warned, no?